### PR TITLE
feat: prefill subdocs

### PR DESCRIPTION
**Pull request by jpangelle**
### Description

This pull request prefills the subdoc agreements with data from the Securitize database if the user is verified and onboarding to the following pools:
- Reif
- RWA/Aave
- New Silver
- 1754
- Consol Freight
- Fortuna Fi S1
- Harbor Trade
- Gig
- 1754 Series 3
- Future new template tbd

### Approvals

- [x] Dev
- [ ] Product

### CODE REVIEW

File: onboarding/api/src/app.controller.ts [hunk 1]
```diff
@@ -21,8 +21,8 @@ export class AppController {
 
   @Get('docusign/callback')
   async getDocusignCallback(): Promise<string> {
-    await this.docusignAuthService.getAccessToken()
-    return 'Callback received'
+    const accessToken = await this.docusignAuthService.getAccessToken()
+    return accessToken
```
Comment (jpangelle) [inline onboarding/api/src/app.controller.ts:25]: Will revert.

File: onboarding/api/src/services/pool.service.ts [hunk 2]
```diff
@@ -31,11 +31,11 @@ export class PoolService {
   }).connect(this.provider)
   registry = new ethers.Contract(config.poolRegistry, contractAbiPoolRegistry, this.provider)
 
-  rwaMarketPermissionManager = new ethers.Contract(
+  rwaMarketPermissionManager = config.rwaMarket.permissionManagerContractAddress ? new ethers.Contract(
     config.rwaMarket.permissionManagerContractAddress,
     contractAbiRwaMarketPermissionManager,
     this.signer
-  )
+  ) : undefined;
```
Comment (jpangelle) [inline onboarding/api/src/services/pool.service.ts:38]: will revert

File: onboarding/api/src/services/templateTabs.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  const templateIds = {
+    REIF: '98cc99ff-8154-4e4c-a500-ba813e8d2a87',
+    'RWA/AAVE': '--',
+    NEW_SILVER: '19750b36-24ab-43e3-9441-23abc2f09a80',
+    1754: 'bc240308-7286-49f4-b1eb-6434a771bc53',
+    CONSOL_FREIGHT: 'a9adc13b-a990-4d58-b54f-647762853923',
+    FORTUNA_FI_S1: '3b3a33b8-23fc-40a4-8a4c-d1f5edd52f96',
+    HARBOR_TRADE: 'fb9d4f62-43b3-4e8f-b135-4d29a4a72a0e',
+    GIG: '--',
+    '1754_SERIES_3': '--',
```
Comment (jpangelle) [inline onboarding/api/src/services/templateTabs.ts:26]: Need to add template id's

File: tinlake-ui/components/Onboarding/KycStep.tsx [hunk 4]
```diff
@@ -15,7 +15,7 @@ interface Props {
 }
 
 // TODO: Redirect to onboard API URL that isn't pool dependant
-const placeholderPoolId = '0x560Ac248ce28972083B718778EEb0dbC2DE55740'
+const placeholderPoolId = '0x1f4155D64D75555B9B36E4514C0FCC11313d5A54'
```
Comment (jpangelle) [inline tinlake-ui/components/Onboarding/KycStep.tsx:18]: will revert

File: tinlake-ui/components/Web3Wallet/Web3Wallet.tsx [hunk 5]
```diff
@@ -31,7 +31,7 @@ export const Web3Wallet: React.FunctionComponent<Props> = ({
   const [copied, setCopied] = useState(false)
   const contRef = useRef<HTMLButtonElement>(null)
   const [, setShowDrop] = useState(false)
-  const { ensName, ensAvatar } = useENS(address)
+  const { ensName, ensAvatar } = {ensName: 'hi', ensAvatar: ''}
```
Comment (jpangelle) [inline tinlake-ui/components/Web3Wallet/Web3Wallet.tsx:34]: will revert

File: tinlake-ui/utils/useOnboardingState.ts [hunk 6]
```diff
@@ -29,7 +29,7 @@ export function useOnboardingState(pool?: Pool | UpcomingPool, tranche?: Tranche
 }
 
 // TODO: Call onboard API URL that isn't pool dependant
-const placeholderPoolId = '0x560Ac248ce28972083B718778EEb0dbC2DE55740'
+const placeholderPoolId = '0x1f4155D64D75555B9B36E4514C0FCC11313d5A54'
```
Comment (jpangelle) [inline tinlake-ui/utils/useOnboardingState.ts:32]: will revert

File: onboarding/api/babel.config.js [hunk 7]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
+  presets: [['@babel/preset-env', { targets: { node: 'current' } }], '@babel/preset-typescript'],
+}
```
Comment (jpangelle) [inline onboarding/api/babel.config.js:3]: config for Jest test transpilation

File: onboarding/api/package.json [hunk 8]
```diff
@@ -33,12 +35,19 @@
...
     "@nestjs/cli": "^7.5.1",
     "@nestjs/schematics": "^7.1.3",
+    "@types/babel__core": "^7",
+    "@types/babel__preset-env": "^7",
     "@types/cookie-parser": "^1",
     "@types/cron": "^1.7.2",
     "@types/express": "^4.17.8",
     "@types/node": "^14.14.10",
+    "babel-jest": "^27.4.6",
+    "jest": "^27.4.7",
```
Comment (jpangelle) [inline onboarding/api/package.json:50]: packages for jest testing

File: onboarding/api/src/services/pool.service.ts (hunk 2 above)
Comment (jpangelle) [inline onboarding/api/src/services/pool.service.ts:38]: @offerijns 
This was a fix to get onboarding api to work locally. Do we want to leave this change in or revert it before merging?

File: tinlake-ui/utils/useENS.ts [hunk 9]
```diff
@@ -10,13 +10,13 @@ const getENSAvatar = async (ensName: string) => {
...
 
-export const useENS = (address: string | null | undefined): ReturnType => {
+export const useENS = (address: string | null | undefined, networkName: string): ReturnType => {
   const [ensName, setENSName] = useState<string | null>(null)
   const [ensAvatar, setENSAvatar] = useState<string | null>(null)
 
   useEffect(() => {
     async function resolveENS() {
-      if (address && ethers.utils.isAddress(address)) {
+      if (networkName === 'Mainnet' && address && ethers.utils.isAddress(address)) {
```
Comment (jpangelle) [inline tinlake-ui/utils/useENS.ts:19]: ENS api's do not work on `Kovan`.

File: onboarding/api/src/services/docusign.service.ts [hunk 10]
```diff
@@ -66,17 +84,25 @@ export class DocusignService {
...
     const recipientViewRequest = {
       authenticationMethod: 'none',
-      email: user.email,
-      userName: user.entityName?.length > 0 ? user.entityName : user.fullName,
+      email: investor.email,
+      userName: investor.details.investorType === 'individual' ? investor.fullName : investor.details.entityName,
       roleName: InvestorRoleName,
       clientUserId: user.id,
-      returnUrl: returnUrl,
+      returnUrl,
```
Comment (jpangelle) [inline onboarding/api/src/services/docusign.service.ts:101]: did this `TODO` while I was here 😄

File: onboarding/api/src/services/pool.service.ts (hunk 2 above)
Comment (jpangelle) [reply onboarding/api/src/services/pool.service.ts:38]: bumping this @offerijns

Comment (offerijns) [reply onboarding/api/src/services/pool.service.ts:38]: This makes sense to leave in imo!

**Review by avinashega — APPROVED** [review]
LGTM

**Pull request merged**